gordo_components/dataset/base.py

```diff
@@ -70,78 +79,87 @@ def join_timeseries(
             index with series-names as top-level and aggregation-method as second-level.
 
         """
-        resampled_series = []
-
+        _series_iterable = []
         for series in series_iterable:
-            startpoint_sametz = resampling_startpoint.astimezone(
-                tz=series.index[0].tzinfo
-            )
-            endpoint_sametz = resampling_endpoint.astimezone(tz=series.index[0].tzinfo)
-
-            if series.index[0] > startpoint_sametz:
-                # Insert a NaN at the startpoint, to make sure that all resampled
-                # indexes are the same. This approach will "pad" most frames with
-                # NaNs, that will be removed at the end.
-                startpoint = pd.Series(
-                    [np.NaN], index=[startpoint_sametz], name=series.name
-                )
-                series = startpoint.append(series)
-                logging.debug(
-                    f"Appending NaN to {series.name} " f"at time {startpoint_sametz}"
+            # Flag to generate pd.Series of NaNs when series` is empty
+            if series.size <= row_threshold:
+                raise InsufficientDataError(
+                    f"Series {repr(series.name)} size ({series.size}) is less than or equal to the minimum required ({row_threshold})."
                 )
 
-            elif series.index[0] < resampling_startpoint:
-                msg = (
-                    f"Error - for {series.name}, first timestamp "
-                    f"{series.index[0]} is before the resampling start point "
-                    f"{startpoint_sametz}"
+            # Resample if we have a resolution set, otherwise simply join the series.
+            if bool(resolution):
```

Comment:
A small thing, but what case are we trying to cover by explicitly calling `bool` here?

Reply:
:+1:  That's another residual from debugging this weird error I was getting.